### DISCUSSION TITLE
Best way to create case-insensitive search

### DISCUSSION THREAD

**Discussion by artemdzyuba** [supabase/supabase#3188] [Questions] [answered]
As far as PostgreSQL has case-sensitive search, what is the best way for creating searchable fields? I use some nocode software with limited SQL queries, so I can't send requests like described on [Stackoverflow](https://stackoverflow.com/questions/7005302/postgresql-how-to-make-case-insensitive-query).

Maybe I can create a generated column to lower the case when addressing this field? Are there any tools specifi to Supabase?

**Comment by steve-chavez** [comment, marked as answer]
Yes, you can create a generated column or a computed column - check an example on this previous discussion https://github.com/supabase/supabase/discussions/2402#discussioncomment-1023391

**Reply by artemdzyuba** [reply]
Thank you!

**Comment by imlutr** [comment]
An alternative solution would be to use [ilike()](https://supabase.com/docs/reference/javascript/ilike). Here is how I used it:
```js
const { data } = await supabaseService
    .from('profiles')
    .select('username')
    .ilike('username', '[username to search]') // Case insensitive search
```

**Reply by theshadowagent** [reply]
I used the same. I wonder how it impacts search efficiency

**Reply by steve-chavez** [reply]
@theshadowagent You can use an index for ilike queries, check https://www.postgresql.org/docs/current/pgtrgm.html#id-1.11.7.42.8

**Reply by theshadowagent** [reply]
> @theshadowagent You can use an index for ilike queries, check https://www.postgresql.org/docs/current/pgtrgm.html#id-1.11.7.42.8

Thanks a lot! I'll use that if I get some performance issues. Could there possibly be an `.ieq()` Supabase function that is faster than `.ilike()` for insensitive equals (I remember that I used such function in Django ORM with PostgresSQL, but not sure if it wasn't ILIKE under the hood)

**Reply by steve-chavez** [reply]
Maybe you refer to this https://docs.djangoproject.com/en/dev/ref/models/querysets/#iexact, from their docs it does use ILIKE under the hood.
